npep/npep-126-egress-traffic-control.md

```diff
@@ -0,0 +1,100 @@
+# NPEP-126: Add northbound traffic support in (B)ANP API
+
+* Issue: [#126](https://github.com/kubernetes-sigs/network-policy-api/issues/126)
+* Status: Provisional
+
+## TLDR
+
+This NPEP proposes adding support for cluster egress (northbound) traffic control in the `AdminNetworkPolicy` and `BaselineAdminNetworkPolicy` API objects.
+
+## Goals
+
+* Implement egress traffic control towards external destinations (outside the cluster)
+* Implement egress traffic control towards cluster nodes
+  - Currently the behaviour for policies defined around traffic from cluster
+    workloads (non-hostNetworked pods) towards nodes in the
+    cluster is undefined. See https://github.com/kubernetes-sigs/network-policy-api/issues/73.
+    - ANP currently supports only east-west traffic and this traffic flow cuts from
+    overlay to underlay which makes this part of the egress (northbound) use case.
+    - Let's provide a defined behaviour in ANP to explicitly achieve the use case.
+    - NOTE1: Traffic towards nodes here includes traffic towards host-networked pods on that node
+      because a "node" resource encompasses all objects that share the host-networking resources
+    - NOTE2: Traffic towards nodes here exluding traffic towards host-networked pods is outside
+      the scope of this NPEP and is a non-goal
+* Implement egress traffic control towards hostNetworked pods
+  - Currently the behaviour for policies defined around traffic from cluster
```

Comment:
Wait I'm confused now is this duplicated?

Reply:
yeah its because we wanted to split the user stories towards the cluster nodes and hostNetworked pods as two?